dataset/src/integrationTest/kotlin/com/cosmotech/dataset/service/DatasetServiceIntegrationTest.kt

```diff
@@ -408,6 +408,67 @@ class DatasetServiceIntegrationTest() : CsmTestBase() {
     assertEquals(DatasetPartTypeEnum.File, inventoryCreatedDatasetPart.type)
   }
 
+  @Test
+  fun `test createDataset with two dataset part with same name`() {
```

Comment:
The test name could be more explicit: "test createDataset with two dataset parts with the same name must fail" :wink:

Reply:
Yes, except it does not test that. It tests the conflict between multipart file names (or dataset source names if I'm wrong. see my other comment). But it's not about the name of the dataset, right?